EXAMPLES.md

```diff
@@ -0,0 +1,73 @@
+# Example
+
+This page shows some examples of converting from ssb-db (and its various helper modules, ssb-backlinks, ssb-query, etc) to ssb-db2.
+
+## Live stream of vote messages
+
+Includes also old messages.
+
+### Before
+
+```js
+pull(
+  sbot.backlinks.read({
+    query: [{ $filter: { dest: msgId } }],
+    index: 'DTA',
+    live: true
+  }),
+  pull.drain(msg => {
+    // ...
+  })
+)
+```
+
+### After
+
+```js
+const {and, votesFor, live, toPullStream} = require('ssb-db2/operators')
+
+pull(
+  sbot.db.query(
+    and(votesFor(msgId)),
+    live({ old: true }),
+    toPullStream()
+  ),
+  pull.drain(msg => {
+    // ...
+  })
+)
+```
+
+## Get latest contact message from Alice about Bob
+
+### Before
+
+```js
+pull(
+  sbot.links({
+    source: aliceId,
+    dest: bobId,
+    rel: 'contact',
+    live: false,
+    reverse: true
+  }),
+  pull.take(1),
+  pull.drain(msg => {
+    // ...
+  })
+)
+```
+
+### After
+
+```js
+const {and, author, contact, descending, toCallback} = require('ssb-db2/operators')
+
+sbot.db.query(
+  and(author(aliceId), contact(bobId)),
+  descending(),
```

Comment:
1st question: yes. We think `ascending` / `descending` are better than `reverse` because `reverse` doesn't tell you what the original ordering would be, it just tells you what the opposite would be.

2nd question: all of these `query` results return messages, not attributes, so far there is no "sort by `___`", all of the messages are sorted by min timestamp.

3rd question: what arj said. I have to correct this example

Reply:
@mixmix my take on descending vs reverse is that we are aiming for an API here that is closer to what people with a bit of SQL experience would expect

Reply:
Yeah ascending / descending sounds like a good clarification.

I think you are sorting here. You're doing ascending by timestamp if I understood.

Is it asserted timestamp or received timestamp or were you meaning "lowest of all timestamps"

Reply:
Bump

Reply:
Its timestamp (Math.min(arrivalTimestamp, declaredTimestamp)) yes. I'll update the docs in jitdb to make this clear. Thanks